dependencies/README.md

```diff
@@ -0,0 +1,112 @@
+  to use appropriate versions of libraries to avoid runtime errors.
+
+### Approach
+
+1. Given an input, the tool prepares a _linkage classpath_, from which the tool inspects Java
+  classes.
+
+2. The tool extracts all _references_ from the class files in the classpath.
+
+3. The tool checks the linkage compatibility of the references through the classpath.
+  
+4. The tool traverses a _class usage graph_ to annotate the linkage errors with _reachability_.
+
+5. At the end, the tool outputs the report on linkage errors.
+
+### Output
+
+- **Static Linkage Errors** are reported by the tool. A static linkage error is either
+  a _dangling reference_ or a _static linkage conflict_.
+
+  A _dangling reference_ is a reference where its destination class does not exist in the 
+  linkage classpath. This error happens when a class is removed in a different version of a library
+  , or there is a dependency missed when constructing the linkage classpath.
+  
+  A _static linkage conflict_ is described in [glossary.md](../library-best-practices/glossary.md).
+  In short, this error happens when the destination class's implementation is not compatible from
+  what it was available at the time of compilation of the source class, in a way that such 
+  incompatibility causes `NoSuchFieldException`, `MethodNotFoundException`, or `LinkageError` when
+  Java Virtual Machine tries to use the implementation.
+
+
+### Class Usage Graph and Reachability
+
+- **Linkage Classpath** is a list of jar files from which the tool inspects Java classes when
+  running the tool against a project. A linkage classpath corresponds to the transitive dependencies
+  induced by the direct dependencies in the target project's pom.xml file. This includes the
+  dependencies marked as `provided` or `optional: true`.
+
+- **Class Usage Graph**
+
+  In order to provide better diagnosis on the output report, the tool builds _class usage
+  graphs_. The nodes (vertex) of the graph correspond to Java classes and the directed edges are
+  references between classes. For example, when 'Class A' has reference to 'Class B' on calling
+  a method X, the tool builds a graph holding an edge between two nodes:
+
+    [Class A] --(method X of class B)-> [Class B]
+
+  In this case, 'Class A' is called the _source class_ of the reference and 'Class B' is called
+  the _destination class_.
+
+- **A reference** between two classes is either _class reference_, _method reference_
+  or _field reference_.
+
+  A _method reference_ indicates the source class uses the (static or non-static) method of
+  the destination class.
+
+  A _field reference_ indicates the source class accesses the (static or non-static) field of
+  the destination class.
+
+  A _class reference_ indicates the source class uses the destination class without specific field 
+  or method (e.g., inheritance).
+
+  A reference is called _linkage conflicting_ when there is a linkage conflict
```

Comment:
"linkage conflicting" feels grammatically odd to me.

Reply:
How about "A reference is called to have a linkage conflict" ?

Reply:
In that case, the sentence would become self-referential, and might as well be removed.